### PR TITLE
MULE-19506: Support new SDK API License annotation package

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaExtensionModelParserUtils.java [hunk 1]
```diff
@@ -268,6 +272,53 @@ private static void checkAnnotationsNotUsedMoreThanOnce(List<? extends Extension
...
+                                org.mule.sdk.api.annotation.license.RequiresEntitlement.class,
+                                requiresEntitlementAnnotation -> new RequiresEntitlementInfo(requiresEntitlementAnnotation.name(),
+                                                                                             requiresEntitlementAnnotation
+                                                                                                 .description()),
+                                requiresEntitlementAnnotation -> new RequiresEntitlementInfo(requiresEntitlementAnnotation.name(),
+                                                                                             requiresEntitlementAnnotation
+                                                                                                 .description()));
+  }
+
+  private static <R extends Annotation, S extends Annotation, T> Optional<T> getInfoFromExtension(ExtensionElement extensionElement,
```
Comment (dabreu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaExtensionModelParserUtils.java:295]: nice function

**Review by dabreu — DISMISSED** [review]

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/info/RequiresEnterpriseLicenseInfo.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.parser.java.info;
+
+
+/**
+ * ADD JDOC
```
Comment (Bardammu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/info/RequiresEnterpriseLicenseInfo.java:11]: Forgot this jdoc :)

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/info/RequiresEntitlementInfo.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.parser.java.info;
+
+
+/**
+ * ADD JDOC
```
Comment (Bardammu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/info/RequiresEntitlementInfo.java:11]: Forgot

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/info/RequiresEnterpriseLicenseInfo.java (hunk 2 above)
Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/info/RequiresEnterpriseLicenseInfo.java:11]: 😢  I am going to add it in a moment

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/info/RequiresEntitlementInfo.java (hunk 3 above)
Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/info/RequiresEntitlementInfo.java]: 🤦‍♂️

**Review by Bardammu — APPROVED** [review]

**Pull request merged**